File: Python/Day09_Palindromepyramid.py
"""
Day 09: Palindrome Pyramid
Printing numbers in a triangular shape that read the same from both sides.
Logic: Spaces -> Decreasing numbers -> Increasing numbers.
"""

n = int(input("Enter the value of n: "))

for i in range(1, n + 1):
    # Printing leading spaces for center alignment
    print(" " * (n - i), end="")
    
    # First half: Numbers decreasing from i down to 1
    for j in range(i, 0, -1):
        print(j, end="")
        
    # Second half: Numbers increasing from 2 up to i
    for j in range(2, i + 1):
        print(j, end="")
    
    # Move to the next line after each row
    print()

File: Java/Day09_Palindromepyramid.java

/**
 * Day 09: Palindrome Pyramid
 * This one is cool! It prints numbers that look the same forward and backward.
 * We use spaces for alignment, then count down, then count up.
 */
import java.util.*;

public class Day09_Palindromepyramid {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the value of n:");
        int n = sc.nextInt();

        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n - i; j++) {
                System.out.print(" ");
            }
            for (int j = i; j >= 1; j--) {
                System.out.print(j);
            }
            for (int j = 2; j <= i; j++) {
                System.out.print(j);
            }
            System.out.println();
        }
    }

}
